core/src/main/java/hudson/Functions.java

```diff
@@ -1302,6 +1304,24 @@ public static ThreadGroupMap sortThreadsAndGetGroupMap(ThreadInfo[] list) {
         return sorter;
     }
 
+    @Extension
+    public static class GlobalConfigHiddenByPermissionFilter extends DescriptorVisibilityFilter {
+
+        @SuppressWarnings("rawtypes")
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+
+            try {
```

Comment:
this should use `hasPermission`

but this filter is global and should be used everywhere which I'm not sure is correct.